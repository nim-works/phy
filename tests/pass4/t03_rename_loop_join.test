discard """
  description: "Ensure that renames joining at a loop start work."
"""
(TypeDefs
  (Int 8)
  (ProcTy (Void))
)
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 1) (Continuations
    (Continuation (Params)
      (Locals (Type 0))
      (Continue 1
        (List (Rename (Local 0))))
    )
    (Continuation (Params (Type 0)) (Locals)
      (SelectBool (IntVal 0)
        (Continue 3 (List))
        (Continue 2 (List))
      )
    )
    (Continuation (Params)
      (Locals (Type 0))
      (Loop 1
        (List (Rename (Local 0))))
    )
    (Continuation (Params) (Locals)
      (Continue 4 (List))
    )
    (Continuation (Params)))
  )
)
