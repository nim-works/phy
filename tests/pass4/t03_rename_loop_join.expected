;$sexp
(TypeDefs
  (Int 8)
  (ProcTy (Void))
)
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 1)
    (Locals (Type 0))
    (Continuations
      (Continuation (Params)
        (Locals (Local 0))
        (Continue 1)
      )
      (Continuation (Params)
        (Locals (Local 0))
        (SelectBool (IntVal 0)
          (Continue 3)
          (Continue 2)
        )
      )
      (Continuation (Params)
        (Locals (Local 0))
        (Loop 1)
      )
      (Continuation (Params) (Locals)
        (Continue 4)
      )
      (Continuation (Params))
    )
  )
)
